docker/package/package_generator.py

```diff
@@ -66,113 +98,80 @@ def get_build_deps(args):
-parser.add_argument("--package", help="specify binary to package")
-parser.add_argument(
-    "--output-dir",
-    help="provide a directory to place the built packages",
-    default="out",
-)
-parser.add_argument(
-    "--binaries-dir",
-    help="provide a directory with exiting prebuilt binaries",
-    type=os.path.abspath,
-)
-parser.add_argument(
-    "--build-sapling-package",
-    help="whether to build the sapling-params package",
-    action="store_true",
-)
-parser.add_argument(
-    "--sources", help="specify source archive for single ubuntu package"
-)
-parser.set_defaults(build_sapling_package=False)
-args = parser.parse_args()
-
-if args.os == "ubuntu":
-    is_ubuntu = True
-elif args.os == "fedora":
-    is_ubuntu = False
-else:
-    raise Exception(
-        "Unexpected package target OS, only 'ubuntu' and 'fedora' are supported."
-    )
-
-if args.type == "source":
-    is_source = True
-elif args.type == "binary":
-    is_source = False
-else:
-    raise Exception(
-        "Unexpected package format, only 'source' and 'binary' are supported."
-    )
-
-package_to_build = args.package
-source_archive = args.sources
-
-if is_ubuntu:
-    run_deps = get_ubuntu_run_deps(args)
-else:
-    run_deps = get_fedora_run_deps(args)
-
-build_deps = get_build_deps(args)
-
-common_deps = run_deps + build_deps
-
-ubuntu_versions = [
-    "bionic",  # 18.04
-    "focal",  # 20.04
-    "jammy",  # 22.04
-]
-
-pwd = os.getcwd()
-home = os.environ["HOME"]
-
-packages_to_build = packages
-
-if args.build_sapling_package:
-    packages.append(sapling_package)
-
-for package in packages_to_build:
-    if package_to_build is None or package.name == package_to_build:
-        if is_ubuntu:
+    args = parser.parse_args()
+
+    is_ubuntu = args.os == "ubuntu"
+
+    is_source = args.type == "source"
+
+    source_archive = args.sources
+
+    binaries_dir = args.binaries_dir
+
+    if is_ubuntu:
+        run_deps = get_ubuntu_run_deps(binaries_dir)
+    else:
+        run_deps = get_fedora_run_deps(binaries_dir)
+
+    build_deps = get_build_deps(binaries_dir)
+
+    common_deps = run_deps + build_deps
+
+    home = os.environ["HOME"]
+
+    os.makedirs(output_dir, exist_ok=True)
+
+    distributions = list(args.distributions)
+
+    packages = []
+
+    for package_name in args.packages:
+        packages.append(all_packages[package_name])
+
+    if is_ubuntu:
+        for package in packages:
             build_ubuntu_package(
                 package,
-                ubuntu_versions,
+                distributions,
                 common_deps,
                 is_source,
                 source_archive,
-                args.binaries_dir,
+                binaries_dir,
             )
+
+        if not is_source:
+            exts = [".deb"]
         else:
+            exts = [".orig.tar.gz", ".dsc", ".changes", ".debian.tar.xz", ".buildinfo"]
+
+        artifacts_dir = "."
+
+    else:
```

Comment:
Here too it's worth check if "fedora" was passed as an argument instead

Reply:
Okay, but it's intended that no validation made in `package_generator.py`

Reply:
that's why no `else` clause